src/components/Autocomplete/components/ComboBox/ComboBox.tsx

```diff
@@ -430,21 +434,26 @@ export default class ComboBox extends React.PureComponent<Props, State> {
   };
 
   private selectOptionAtIndex = (newOptionIndex: number) => {
-    const {navigableOptions, selectedOption: oldSelectedOption} = this.state;
-    if (!navigableOptions || navigableOptions.length === 0) {
-      return;
-    }
-    const newSelectedOption = navigableOptions[newOptionIndex];
+    this.setState((prevState) => {
```

Comment:
This change helps fix the selected index skipping issue by making sure when we use the previous to determine new state, its in the setState function.